src/main/scala/scorex/core/ModifiersCache.scala

```diff
@@ -50,11 +50,13 @@ trait ModifiersCache[PMOD <: PersistentNodeViewModifier, H <: HistoryReader[PMOD
   def contains(key: K): Boolean = cache.contains(key) || rememberedKeys.contains(key)
 
   def put(key: K, value: V): Unit = synchronized {
-    onPut(key)
-    cache.put(key, value)
-    currentCacheSize = currentCacheSize + 1
+    if(!contains(key)) {
+      onPut(key)
```

Comment:
Can there ever be a case when you might want `onPut` to be called even when the cache already contains the key?